### PR TITLE
IOS Fix -  NSCocoaErrorDomain error 256

### PR DESCRIPTION
**Pull request by EduFrazao**
Fixed a IOS Bug, throwing NSCocoaErrorDomain code 256 whein trying to read absolutePath's of files ready to be uploaded.

### CODE REVIEW

**Comment by edeckers** [comment]
Thank you @EduFrazao ! I'll try to take a look and merge this weekend

**Comment by EduFrazao** [comment]
> Thank you @EduFrazao ! I'll try to take a look and merge this weekend
Thank you @edeckers !!!

**Comment by edeckers** [comment]
This is taking longer than expected, as I've had some issues with the builder machine not being able to kill the test runs correctly. Also, some iOS-tests are failing and I didn't have time to investigate yet. I hope to find some time for this soon

**Comment by EduFrazao** [comment]
> This is taking longer than expected, as I've had some issues with the builder machine not being able to kill the test runs correctly. Also, some iOS-tests are failing and I didn't have time to investigate yet. I hope to find some time for this soon

Hi @edeckers ! Thanks for your answer!
No problem, I'm running a patched version on my application waiting for your review. Thank you very mutch.

**Comment by edeckers** [comment]
tl; dr: found the problem that causes the tests to fail, did not have time to fix yet

--

I had some time at my disposal, so I was able to track down the issue with regards to the broken tests.

The `buildRequestDataForFileUpload` receives a `parts` parameter, which contains - among other things  - the textual representation of the path to the local file to be uploaded.

When this path is prefixed with `file://` schema in the input, e.g. `file:///path/to/your/file.ext` this causes problems, because `URL(fileURLWithPath: ...` replaces the `file://` schema with `file:/`, e.g. `file:/path/to/your/file.ext` which results in a 'file not found' exception.

There are some instant gratification hacks to circumvent this problem, like stripping `file://` from the provided url, but that's just that: a hack, which will probably open a whole other can of worms and is not something that I'd like to introduce to the codebase.

So, I will need to find some time again to fix it properly, but the available time is sparse these days so it might be a while. If you're up to it, I'd happily accept a PR that addresses the situation!

**Comment by edeckers** [comment]
Fixed in #265
